src/common/Requirements.ts

```diff
@@ -120,38 +122,79 @@ export class RequirementProcessor {
-            tests: []
-        };
-
-        let totalDuration = 0;
-        let enabledRequirements: Requirement[] = [];
-
+        const requirementCheckResults: RequirementCheckResult[] = [];
+        const tasksToRun: Listr[] = [];
         // eslint-disable-next-line @typescript-eslint/no-unused-vars
         Object.entries(requirements).forEach(([_, requirementList]) => {
             if (requirementList.enabled) {
-                enabledRequirements = enabledRequirements.concat(
-                    requirementList.requirements
+                const requirementCheckResult: RequirementCheckResult = {
+                    hasMetAllRequirements: true,
+                    tests: []
+                };
+                requirementCheckResults.push(requirementCheckResult);
+                tasksToRun.push(
+                    this.createTasks(requirementList, requirementCheckResult)
                 );
             }
         });
 
-        if (enabledRequirements.length === 0) {
+        try {
+            await this.processTasks(tasksToRun);
+
+            let hasMetAllRequirements = true;
+
+            for (const results of requirementCheckResults) {
+                if (!results.hasMetAllRequirements) {
+                    hasMetAllRequirements = false;
+                }
+            }
+
+            if (!hasMetAllRequirements) {
+                return Promise.reject(
+                    new SfdxError(
+                        util.format(
+                            messages.getMessage('error:requirementCheckFailed'),
+                            platform
+                        ),
+                        'lwc-dev-mobile-core',
+                        [
+                            messages.getMessage(
+                                'error:requirementCheckFailed:recommendation'
+                            )
+                        ]
+                    )
+                );
+            }
+
             return Promise.resolve();
+        } catch (error) {
+            return Promise.reject(
+                new SfdxError(
+                    util.format('unexpected error %s', error),
+                    'lwc-dev-mobile-core'
+                )
+            );
         }
+    }
+
+    private static createTasks(
+        requirementList: RequirementList,
+        testResult: RequirementCheckResult
+    ): Listr {
+        const rootTaskTitle = requirementList.title;
+        let totalDuration = 0;
 
-        const rootTaskTitle = messages.getMessage('rootTaskTitle');
-        const requirementTasks = new Listr(
+        return new Listr(
             {
                 task: (_rootCtx, rootTask): Listr => {
                     const subTasks = new Listr([], {
                         concurrent: true,
                         exitOnError: false
                     });
-                    for (const requirement of enabledRequirements) {
+                    for (const requirement of requirementList.requirements) {
```

Comment:
If I understand your changes correctly, we used to have an array of requirements and now we have an array of array of requirements, right? If so then you need to update your logic in lines 203-206 to (a) make the current array as `hasMetAllRequirements = false` and also mark the root array (containing all other arrays) with `hasMetAllRequirements = false`

Reply:
The logic to resolve requirements will resolve each main task separately.  It waits for each task (including all subtasks) to resolve before checking the next task.  And will check the array of results before proceeding.  See lines 149-152.